### PR TITLE
Make it possible to specify an arbitrary Terraform version

### PR DESCRIPTION
**Pull request by iangelov**
In our environment, we found it useful to have the ability to build container images with arbitrary Terraform versions. To that end, I made the following changes which should allow one to do so.

I also added "-Ss --fail" to the curl command line, so that curl doesn't output progress if everything is working correctly, but do so if there's an error, and fail if necessary. This should produce less garbage output and streamline diagnosing issues. Additionally, the container image is properly tagged with the Terraform version, which allows one to be more explicit about which version should actually be used depending on the use case.

The current implementation allows for the Dockerfile to be used independently from Cloud Build as well.

### CODE REVIEW

File: terraform/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 FROM gcr.io/cloud-builders/gcloud
 
-ENV TERRAFORM_VERSION=0.12.6
-ENV TERRAFORM_VERSION_SHA256SUM=6544eb55b3e916affeea0a46fe785329c36de1ba1bdb51ca5239d3567101876f
+ARG TERRAFORM_VERSION=0.12.6
+ARG TERRAFORM_VERSION_SHA256SUM=6544eb55b3e916affeea0a46fe785329c36de1ba1bdb51ca5239d3567101876f
```
Comment (leeonlee) [inline terraform/Dockerfile]: Removing the default args gives us only one place to manage the default version in the substitutions

File: terraform/cloudbuild.yaml [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+  - --build-arg
+  - TERRAFORM_VERSION
+  - --build-arg
+  - TERRAFORM_VERSION_SHA256SUM
+  - --tag=gcr.io/${PROJECT_ID}/terraform:${_TERRAFORM_VERSION}
+  - .
+- name: 'gcr.io/${PROJECT_ID}/terraform:${_TERRAFORM_VERSION}'
   args: ['version']
-images: ['gcr.io/${PROJECT_ID}/terraform']
-tags: ['cloud-builders-community']
```
Comment (leeonlee) [inline terraform/cloudbuild.yaml]: Why was the tag removed?

Comment (iangelov) [reply terraform/cloudbuild.yaml]: Are you referring to the implicit value of the tag "latest", or to the cloudbuild.yaml tag? I consider the former to be a good practice in general, although not necessarily what's intended in this case. The latter didn't exist at the time of the fork, but I've made sure to add it as it's a great thing to filter on. In any case - let's clarify the intent and I will make the necessary changes.

File: terraform/Dockerfile (hunk 1 above)
Comment (iangelov) [reply terraform/Dockerfile]: I left it in place to facilitate local testing, but it makes sense. I've removed the hardcoded values in the Dockerfile.

**Comment by iangelov** [comment]
@googlebot I fixed it.

File: terraform/cloudbuild.yaml (hunk 2 above)
Comment (leeonlee) [reply terraform/cloudbuild.yaml]: Sorry I wasn't clear, I was referring to the cloud build tag.

**Comment by leeonlee** [comment]
The changes look good to me, just need to fix the CLA check.

**Comment by leeonlee** [comment]
Thanks!

**Comment by iangelov** [comment]
@leeonlee - thank you for reviewing and merging this pull request!

**Pull request merged**